guides-markdown/events.md

```diff
@@ -0,0 +1,261 @@
+| `tx` | [TX](https://github.com/bcoin-org/bcoin/blob/master/lib/primitives/tx.js), [Details](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/txdb.js) | [TXDB](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/txdb.js): `insert()` | _txdb_&#8594;[WalletDB](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/walletdb.js) (also&nbsp;returns&nbsp;Wallet)<br>_txdb_&#8594;[Wallet](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/wallet.js)|
+| `double spend` | [MempoolEntry](https://github.com/bcoin-org/bcoin/blob/master/lib/mempool/mempoolentry.js) | [Mempool](https://github.com/bcoin-org/bcoin/blob/master/lib/mempool/mempool.js): `removeDoubleSpends()` | |
+| `balance` | [Balance](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/txdb.js) | [TXDB](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/txdb.js): `insert()`, `confirm()`, `disconnect()`, `erase()` | _txdb_&#8594;[WalletDB](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/walletdb.js) (also&nbsp;returns&nbsp;Wallet)<br>_txdb_&#8594;[Wallet](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/wallet.js)|
+| `address` | [[WalletKey](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/walletkey.js)] | [Wallet](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/wallet.js): `_add()`, | _wallet_&#8594;[WalletDB](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/walletdb.js) (returns&nbsp;parent&nbsp;Wallet,&nbsp;[WalletKey]) |
+
+## Socket Events
+
+Websocket connections in bcoin are handled by two servers, one for `Node` and one for `Wallet`.
+Those servers each have child objects such as `Chain`, `Mempool`, `Pool`, and `WalletDB`, and
+relay events from them out the socket. To receive an event, the socket client must watch a
+channel (such as `chain`, `mempool`, or `auth`) or join the a wallet
+(which would be user-defined like `primary`, `hot-wallet`, or `multisig1`). All wallets can be
+joined at once by joining `'*'`. 
+
+### Listen for Socket Events
+
+To make a socket connection to bcoin, you need to run a websocket client.
+Luckily the bcoin developers have developed [`bsock`](https://github.com/bcoin-org/bsock), 
+a minimal websocket-only implementation of the socket.io protocol. By default, bsock listens on
+`localhost`, and you only need to pass it a port number to connect to one of the bcoin servers.
+The example below illustrates how to establish the socket connection, authenticate with your
+user-defined [API key](http://bcoin.io/api-docs/#authentication) and then send and recieve events!
+[See the tables below for a complete list of calls and events available in bcoin.](#socket-events-directory)
+
```

Comment:
why have a bsock example rather than a bclient one? connecting with bclient is even more straightforward since it will set the ports automatically based on network.

Reply:
I wanted to show the low level interaction with bcoin socket sever with actual hook, bind, call, etc... The raw dirty mechanics. But I could add another paragraph about how much easier this all is with bclient, which uses bcurl, which uses bsock :-)

Reply:
Yeah, I think that could be nice, especially since in many cases people will already be using bclient in some capacity to do normal http requests so it would be good to show that that can be used for both. I totally agree though that it's nice to see how are things are working at a lower level.